### PR TITLE
improve tiered storage system test

### PR DESCRIPTION
**Pull request by showuon**
### Type of change

_Select the type of your PR_

- Enhancement=
- Documentation

### Description

1. In `testTieredStorageWithAivenPlugin`, we'll wait for the log uploaded or deleted in remote storage, this can be speeded up by reducing `remote.log.manager.task.interval.ms` config value (default is 30 sec). 
2. In `testTieredStorageWithAivenPlugin`, we'll wait for the local log got deleted, this can be speeded up by reducing `log.retention.check.interval.ms` config value (default is 5 min). 
3. Update the dev guide because we don't have `testCustomAndUpdatedValues` test in `KafkaST` now.

### Checklist

_Please go through this checklist and make sure all applicable tasks have been done_

- [ ] Write tests
- [x] Make sure all tests pass
- [x] Update documentation
- [ ] Check RBAC rights for Kubernetes / OpenShift roles
- [ ] Try your changes from Pod inside your Kubernetes and OpenShift cluster, not just locally
- [ ] Reference relevant issue(s) and close them after merging
- [ ] Update CHANGELOG.md
- [ ] Supply screenshots for visual changes, such as Grafana dashboards

### CODE REVIEW

**Review by im-konge — APPROVED** [review]
LGTM, thanks

**Review by henryZrncik — APPROVED** [review]

**Review by see-quick — APPROVED** [review]
LGTM, just out of curiosity how quicker will test go now? (compared to previous runs?) I assumed you have tested it.

**Comment by showuon** [comment]
> LGTM, just out of curiosity how quicker will test go now? (compared to previous runs?) I assumed you have tested it.

@see-quick , I've removed the test terminals. But the most of time in the test is:
1. waiting for data uploaded to S3
2. waiting for local log deletion 
3. waiting for S3 data is empty

For (1), the related config is: `remote.log.manager.task.interval.ms`, whose default is 30 sec. So we might need additional 30 sec if we missed the previous check.
For (2), the topic config (`file.delete.delay.ms=1000`) has fixed it.
For (3), the related config is `log.retention.check.interval.ms`, whose default is 5 min. This has the most influence because if we missed the previous check, the test will need to wait for another 5 min.

So, basically, in the worst case, we can save (1) 30 sec + (3)300 sec - (new 1) 5 sec - (new 3) 5 sec = 320 sec, which is more than 5 mins.

**Comment by see-quick** [comment]
> > LGTM, just out of curiosity how quicker will test go now? (compared to previous runs?) I assumed you have tested it.
> 
> @see-quick , I've removed the test terminals. But the most of time in the test is:
> 
> 1. waiting for data uploaded to S3
> 2. waiting for local log deletion
> 3. waiting for S3 data is empty
> 
> For (1), the related config is: `remote.log.manager.task.interval.ms`, whose default is 30 sec. So we might need additional 30 sec if we missed the previous check. For (2), the topic config (`file.delete.delay.ms=1000`) has fixed it. For (3), the related config is `log.retention.check.interval.ms`, whose default is 5 min. This has the most influence because if we missed the previous check, the test will need to wait for another 5 min.
> 
> So, basically, in the worst case, we can save (1) 30 sec + (3)300 sec - (new 1) 5 sec - (new 3) 5 sec = 320 sec, which is more than 5 mins.

Thanks, and good job with such optimization! 👍

**Comment by scholzj** [comment]
Thanks for the PR.
